### PR TITLE
Add NFT pallet

### PR DESCRIPTION
**Pull request by Satoshi-Kusumoto**
Implemented 3 types of NFT: simple  (traditional issuance and mint), claim(whitelisted user can claim), and merge (merge 2 base NFTs and get this one)
Have one basic test for each NFT type, definitely need more as the next step.

@juniuszhou currently in mock.rs, the advance block is not working, you may have a look and check if can fix this

### CODE REVIEW

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@ name: Rust
 
 on:
   push:
-    branches: [ dev, rococo ]
+    branches: [ dev, stable ]
   pull_request:
-    branches: [ dev, rococo ]
+    branches: [ dev, stable ]
```
Comment (h4n0) [inline .github/workflows/build_test.yml:7]: Will we abandon rococo? If yes we need to fix CI dependencies on parachain repo as well.

**Comment by suinuj** [comment]
> @juniuszhou currently in mock.rs, the advance block is not working, you may have a look and check if can fix this

will check it.

**Comment by Satoshi-Kusumoto** [comment]
> > @juniuszhou currently in mock.rs, the advance block is not working, you may have a look and check if can fix this
> 
> will check it.

You may fix it latter

File: pallets/nft/src/weights.rs [hunk 2]
```diff
@@ -0,0 +1,122 @@
+// This file is part of Acala.
+
+// Copyright (C) 2020-2021 Acala Foundation.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
```
Comment (h4n0) [inline pallets/nft/src/weights.rs]: Please update this part.

**Comment by suinuj** [comment]
I fixed the advance block with a commit.

File: .github/workflows/build_test.yml (hunk 1 above)
Comment (Satoshi-Kusumoto) [reply .github/workflows/build_test.yml:7]: Yes rococo is abandoned

File: pallets/nft/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+					if TokenIdOf::<T>::from(quantity) + issued > TokenIdOf::<T>::from(max_num) {
+						Err(Error::<T>::QuantitiyOverflow)?
+					}
+				}
+				_ => {
+					Err(Error::<T>::WrongClassType)?
+				}
+			}
+
+			let data = TokenData {
```
Comment (h4n0) [inline pallets/nft/src/lib.rs:250]: Maybe you forget to add TODO here like below what you did. I added it in order not to be forgotten.
```suggestion
                         // TODO: adjustible rarity
			let data = TokenData {
```

**Review by h4n0 — APPROVED** [review]

**Pull request merged**